app/components/avo/fields/boolean_field/edit_component.html.erb

```diff
@@ -1,13 +1,47 @@
 <%= field_wrapper(**field_wrapper_args) do %>
   <div class="h-8 flex items-center">
-    <%= @form.check_box @field.id,
-      value: @field.value,
-      checked: @field.value,
-      class: "text-lg h-4 w-4 checked:bg-primary-400 focus:checked:!bg-primary-400 rounded #{@field.get_html(:classes, view: view, element: :input)}",
-      data: @field.get_html(:data, view: view, element: :input),
-      disabled: disabled?,
-      autofocus: @autofocus,
-      style: @field.get_html(:style, view: view, element: :input)
+    <% 
+      # Common checkbox attributes
+      common_checkbox_attrs = {
+        value: @field.value,
+        checked: @field.value,
+        disabled: disabled?,
+        autofocus: @autofocus,
+        id: "#{@field.id}_checkbox"
+      }
     %>
+    
+    <% if @field.instance_variable_get(:@args)&.dig(:as_toggle) %>
+      <%# Stimulus controller wrapper %>
+      <div data-controller="avo-boolean-toggle">
+        <%= @form.check_box @field.id, 
+            class: "sr-only", 
+            data: { 
+              **@field.get_html(:data, view: view, element: :input),
+              avo_boolean_toggle_target: "checkbox"
+            },
+            **common_checkbox_attrs %>
```

Comment:
Let's aim for a single `check_box` declaration, with helper methods in `app/components/avo/fields/boolean_field/edit_component.rb` that return the appropriate classes and data attributes based on the `as_toggle` condition.